mobile/test/common/integration/test_server.cc

```diff
@@ -58,24 +59,28 @@ TestServer::TestServer()
       .WillByDefault(testing::ReturnRef(*stats_store_.rootScope()));
 }
 
-void TestServer::startTestServer(bool use_quic) {
+void TestServer::startTestServer(bool use_quic, bool disable_https) {
   ASSERT(!upstream_);
   // pre-setup: see https://github.com/envoyproxy/envoy/blob/main/test/test_runner.cc
   Logger::Context logging_state(spdlog::level::level_enum::err,
                                 "[%Y-%m-%d %T.%e][%t][%l][%n] [%g:%#] %v", lock, false, false);
   // end pre-setup
-
+  ASSERT(!(use_quic && disable_https), "test_server doesn't support QUIC without https");
+  Network::DownstreamTransportSocketFactoryPtr factory;
   if (use_quic) {
     upstream_config_.upstream_protocol_ = Http::CodecType::HTTP3;
     upstream_config_.udp_fake_upstream_ = FakeUpstreamConfig::UdpConfig();
+    factory = createQuicUpstreamTlsContext(factory_context_);
   } else {
```

Comment:
nit: let's avoid a layer of indenting by making this `} else if (disable_https) {`

Reply:
Resolved via the enum refactor